pkg/sql/alter_table.go

```diff
@@ -906,11 +905,73 @@ func applyColumnMutation(
 				col.Type.SQLString(), typ.SQLString())
 		case schemachange.ColumnConversionTrivial:
 			col.Type = *typ
-		default:
-			return unimplemented.NewWithIssueDetail(9851,
-				fmt.Sprintf("%s->%s", col.Type.SQLString(), typ.SQLString()),
-				"type conversion not yet implemented")
-		}
+
+		default: // try conversion
+			//1. Create a new computed column c' using c::t'
+			// Computed off old column, need to update ComputeExpr
+			// Make shadow column for now
+
+			// Can concurrent schema changes be happening?
+			currentMutationID := tableDesc.ClusterVersion.NextMutationID
+			for i := range tableDesc.Mutations {
+				mut := &tableDesc.Mutations[i]
+				if mut.MutationID < currentMutationID {
+					return unimplemented.Newf(
+						"column type change while undergoing schema change",
+						"table %s is currently undergoing a schema change", tableDesc.Name)
+				}
+			}
+
+			nameExists := func(name string) bool {
+				_, _, err := tableDesc.FindColumnByName(tree.Name(name))
+				return err == nil
+			}
+
+			name := sqlbase.GenerateUniqueConstraintName(
```

Comment:
maybe put in a new name that is clear that this is the column that is replacing the new column.